### DISCUSSION TITLE
How to replicate ActivePivot's doNotAggregateAbove()?

### DISCUSSION THREAD

**Discussion by EricDallAgnol** [atoti/atoti#609] [Q&A] [answered]
Hi 👋 

I would like to implement a measure that would aggregate from a certain level of granularity but not to the top. Basically, to have a `doNotAggregateAbove()`, as it is possible with [CopperAPI](https://www.activeviam.com/activepivot/latest/docs/copper/copper_measures/#aggregation-levels-hiding)
Is there a way to do such behaviour with the functions currently exposed by the PythonAPI ? If no, is it planned ?

Many thanks

**Comment by CamilleBlanco** [comment]
Hi 😄 

This feature is not implemented in the Python API, and it is not planned to add it at the moment. Though we are interested in hearing more about your use case: would you mind explaining it and why you would appreciate to have this option? 😃 

Thank you !!

**Comment by EricDallAgnol** [comment]
Hi Camille,

Thanks for your answer.
Here is basically my use case :
## Context : Value-at-Risk

## Input : 
  - `VaR`: we are receiving VaR data on two levels of granularity : Trade and/or Leg. Our VaR is defined as an `tt.type.DOUBLE_ARRAY`, with a lenght N
  - `Scenario`: we are receiving another input and among it we have what we call a ScenarioNames which is basically a String array, with a lenght N too

## Goal :
  - For each index of our VaR vector we associate a ScenarioName and we want to aggregate at this level but not above, see below :

## Example:

Dummy data : 
 VaR (we consider here only the VaR value received for specific Trade / Leg at index X)
- Trade 0 / Leg 0 -> 881.16
- Trade 1 / Leg 1 -> 881.16
- Trade 2 / Leg2_0 -> 881.16
- Trade 2 / Leg2_1 -> 881.16

And we associate this index X (ie these values) to the ScenarioName "20210315" (I skip this part given it's not relevant)

![image](https://user-images.githubusercontent.com/83015366/173574903-79a733e7-5654-48d8-a7af-6078d03b63e8.png)

We need to perform an aggregation but it only makes sense to aggregate from Trade / Leg and not above our ScenarioName

Let me know if this is clear enough for you
Many thanks

**Comment by CamilleBlanco** [comment, marked as answer]
Hi!

Many thanks for the details 🚀 

Here, a way to avoid the aggregation above the `ScenarioNames` level would be to define your measure as `tt.where(l["ScenarioNames"]!=None, m["VaR.vector"], None)`.

Also, if it is a fixed vector, `ScenarioNames` could perhaps be a parameter hierarchy, roughly:
```python
[...]
cube.create_parameter_hierarchy_from_members(
    "ScenarioNames",
    ["20210315", "20210316", "20210317"],
    index_measure_name="Scenario index",
)
cube.measures["VaR.vector"]=tt.value(table["VaR"])[cube.measures["Scenario index"]]
```
This would avoid the aggregation above `ScenarioNames` since a parameter hierarchy is slicing, yet I don't know if it is accurate in the complete context of your use case. If not, the `tt.where` solution should work anyway 😄 

Let me know if this answers your question! 
Thank you

**Reply by EricDallAgnol** [reply]
Hi Camille,

Indeed, this `tt.where` meets perfectly my expectation !
_(No it's not a fixed vector thus it doesn't make sense in my case to make it slicing_)

Many thanks for your answer
